bin/desi_qso_qn_afterburner

```diff
@@ -0,0 +1,376 @@
+                                               ('Z_LYA', 'f4'), ('Z_CIV', 'f4'), ('Z_CIII', 'f4'), ('Z_MgII', 'f4'), ('Z_Hbeta', 'f4'), ('Z_Halpha', 'f4')])
+
+    data['TARGETID'] = dataframe['TARGETID']
+    data['RA'] = dataframe['RA']
+    data['DEC'] = dataframe['DEC']
+
+    data['Z_NEW'] = dataframe['Z_NEW']
+    data['ZERR_NEW'] = dataframe['ZERR_NEW']
+    data[DESI_TARGET] = dataframe[DESI_TARGET]
+    data['COEFFS'] = np.array([np.array(dataframe['COEFFS'][i]) for i in range(dataframe.shape[0])])
+    data['SPECTYPE'] = dataframe['SPECTYPE']
+    data['Z_RR'] = dataframe['Z_RR']
+    data['IS_QSO_QN_NEW_RR'] = dataframe['IS_QSO_QN_NEW_RR']
+
+    data['Z_QN'] = dataframe['Z_QN']
+    data['C_LYA'] = np.array([dataframe['C_LINES'][i][0] for i in range(dataframe.shape[0])])
+    data['C_CIV'] = np.array([dataframe['C_LINES'][i][1] for i in range(dataframe.shape[0])])
+    data['C_CIII'] = np.array([dataframe['C_LINES'][i][2] for i in range(dataframe.shape[0])])
+    data['C_MgII'] = np.array([dataframe['C_LINES'][i][3] for i in range(dataframe.shape[0])])
+    data['C_Hbeta'] = np.array([dataframe['C_LINES'][i][4] for i in range(dataframe.shape[0])])
+    data['C_Halpha'] = np.array([dataframe['C_LINES'][i][5] for i in range(dataframe.shape[0])])
+
+    data['Z_LYA'] = np.array([dataframe['Z_LINES'][i][0] for i in range(dataframe.shape[0])])
+    data['Z_CIV'] = np.array([dataframe['Z_LINES'][i][1] for i in range(dataframe.shape[0])])
+    data['Z_CIII'] = np.array([dataframe['Z_LINES'][i][2] for i in range(dataframe.shape[0])])
+    data['Z_MgII'] = np.array([dataframe['Z_LINES'][i][3] for i in range(dataframe.shape[0])])
+    data['Z_Hbeta'] = np.array([dataframe['Z_LINES'][i][4] for i in range(dataframe.shape[0])])
+    data['Z_Halpha'] = np.array([dataframe['Z_LINES'][i][5] for i in range(dataframe.shape[0])])
+
+    fits = fitsio.FITS(filename, 'rw', clobber=clobber)
+    fits.write(data, extname='QN+RR')
+    if clobber:
+        log.info(f'OVERWRITE the file : {filename}')
+    else:
+        log.info(f'EXPAND the file : {filename}')
+    fits.close()
+
+    return
+
+
+if __name__ == "__main__":
+
+    args = collect_argparser()
+
+    ## Selection param for QuasarNet
+    param_QN = {'c_thresh':args.c_thresh, 'n_thresh':args.n_thresh}
+
+    param_RR = {'template_filename':args.template,
+                'filename_priors':args.filename_priors,
+                'filename_output_rerun_RR':args.filename_output_rerun_RR,
+                'filename_zbest_rerun_RR':args.filename_zbest_rerun_RR,
+                'delete_RR_output':args.delete_RR_output}
+
+    if os.path.isfile(args.coadd) & os.path.isfile(args.zbest):
+        # open best fit file generated by redrock
+        zbest_file = fitsio.FITS(args.zbest)
+        zbest = zbest_file[1]
+        fibermap = zbest_file[2]
+        index_sorted = np.argsort(fibermap['TARGETID'][:500]) # in zbest targetid are sorted, it is not the case in the fibermap
```

Comment:
See comments in desi_qso_mgii_afterburner about using `np.unique(..., return_index=True)` to be more robust for the future, as well as using `desitarget.targets.main_cmx_or_sv` to derive DESI_TARGET.

Reply:
ok